graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/hooks/SchemaGeneratorHooks.kt

```diff
@@ -90,22 +100,58 @@ interface SchemaGeneratorHooks {
     /**
      * Called after wrapping the type based on nullity but before adding the generated type to the schema
      */
-    fun didGenerateGraphQLType(type: KType, generatedType: GraphQLType) = generatedType
+    @Suppress("Detekt.ThrowsCount")
+    fun didGenerateGraphQLType(type: KType, generatedType: GraphQLType): GraphQLType {
+        val unwrapped = GraphQLTypeUtil.unwrapNonNull(generatedType)
+        return when {
+            unwrapped is GraphQLInterfaceType && unwrapped.fieldDefinitions.isEmpty() -> throw EmptyInterfaceTypeException(ktype = type)
+            unwrapped is GraphQLObjectType && unwrapped.fieldDefinitions.isEmpty() -> throw EmptyObjectTypeException(ktype = type)
+            unwrapped is GraphQLInputObjectType && unwrapped.fieldDefinitions.isEmpty() -> throw EmptyInputObjectTypeException(ktype = type)
+            else -> generatedType
+        }
+    }
 
     /**
      * Called after converting the function to a field definition but before adding to the schema to allow customization
      */
-    fun didGenerateQueryType(kClass: KClass<*>, function: KFunction<*>, fieldDefinition: GraphQLFieldDefinition): GraphQLFieldDefinition = fieldDefinition
+    fun didGenerateQueryFieldType(kClass: KClass<*>, function: KFunction<*>, fieldDefinition: GraphQLFieldDefinition): GraphQLFieldDefinition = fieldDefinition
```

Comment:
Do we need to have `type` as a suffix for all of these hooks, or do we just want to call them `didGenerate*Field` `didGenerate*Object`?

Reply:
I just followed the naming of other hooks but can change if it needed

Reply:
If we are going to make a breaking change, I think we should drop the `Type` from this names. It makes sense on the other ones because it referencing the `GraphQLType`, but here `QueryObjectType` is a bit redundant to `QueryObject`